projects/packages/publicize/src/class-rest-controller.php

```diff
@@ -344,4 +413,106 @@ public function make_proper_response( $response ) {
 	protected function get_blog_id() {
 		return $this->is_wpcom ? get_current_blog_id() : Jetpack_Options::get_option( 'id' );
 	}
+
+	/**
+	 * Calls the WPCOM endpoint to update the publicize connection.
+	 *
+	 * POST jetpack/v4/social/connections/{connection_id}
+	 *
+	 * @param WP_REST_Request $request The request object, which includes the parameters.
+	 */
+	public function update_publicize_connection( $request ) {
+		$external_user_id = $request->get_param( 'external_user_ID' );
+		$shared           = $request->get_param( 'shared' );
+		$blog_id          = $this->get_blog_id();
+		$connection_id    = $request->get_param( 'connection_id' );
+
+		$path = sprintf(
+			'/sites/%d/jetpack-social-connections/%d',
+			$blog_id,
+			$connection_id
+		);
+
+		$body = array();
+
+		if ( ! empty( $external_user_id ) ) {
+			$body['external_user_ID'] = $external_user_id;
+		}
+
+		if ( $shared || ( false === $shared ) ) {
+			$body['shared'] = $shared;
+		}
+
+		$response = Client::wpcom_json_api_request_as_user(
+			$path,
+			'2',
+			array(
+				'method'  => 'POST',
+				'timeout' => 120,
+			),
+			$body,
+			'wpcom'
+		);
+		return rest_ensure_response( $this->make_proper_response( $response ) );
+	}
+
+	/**
+	 * Calls the WPCOM endpoint to delete the publicize connection.
+	 *
+	 * DELETE jetpack/v4/social/connections/{connection_id}
+	 *
+	 * @param WP_REST_Request $request The request object, which includes the parameters.
+	 */
+	public function delete_publicize_connection( $request ) {
+		$connection_id = $request->get_param( 'connection_id' );
+		$blog_id       = $this->get_blog_id();
+
+		$path = sprintf(
+			'/sites/%d/jetpack-social-connections/%d',
+			$blog_id,
+			$connection_id
+		);
+
+		$response = Client::wpcom_json_api_request_as_user( $path, '2', array( 'method' => 'DELETE' ), null, 'wpcom' );
+		return rest_ensure_response( $this->make_proper_response( $response ) );
+	}
+
+	/**
+	 * Create a publicize connection
+	 *
+	 * @param WP_REST_Request $request The request object, which includes the parameters.
+	 * @return WP_REST_Response|WP_Error True if the request was successful, or a WP_Error otherwise.
+	 */
+	public function create_publicize_connection( $request ) {
+		$keyring_connection_id = $request->get_param( 'keyring_connection_ID' );
+		$shared                = $request->get_param( 'shared' );
+		$external_user_id      = $request->get_param( 'external_user_ID' );
+		$blog_id               = $this->get_blog_id();
+
+		$path = sprintf(
+			'/sites/%d/jetpack-social-connections/new',
```

Comment:
It's probably not that important, but I've realised that we could remove the `/new` here too, in order to make it consistent